src/components/attendee/ScheduleOverlay/ScrollableWeekdays.tsx

```diff
@@ -0,0 +1,188 @@
+  setSelectedDayIndex: React.Dispatch<React.SetStateAction<number>>;
+  selectedDayIndex: number;
+  dayDifference: number;
+  isScheduleTimeshifted: boolean;
+};
+export const ScrollableWeekdays: React.FC<ScrollableWeekDaysProps> = ({
+  setSelectedDayIndex,
+  selectedDayIndex,
+  dayDifference,
+  isScheduleTimeshifted,
+}) => {
+  const { userWithId } = useUser();
+  const userEventIds =
+    userWithId?.myPersonalizedSchedule ?? ALWAYS_EMPTY_OBJECT;
+
+  const { firstScheduleDate, liveAndFutureEvents } = useVenueScheduleEvents({
+    userEventIds,
+  });
+
+  const allRefs = useRef<RefObject<HTMLButtonElement>[]>(
+    range(dayDifference).map(() => React.createRef())
+  );
+  const firstDayIntersect = useIntersection(allRefs?.current?.[0], {
+    rootMargin: "0px",
+  });
+  const lastDayIntersect = useIntersection(
+    allRefs?.current?.[dayDifference - 1],
+    {
+      rootMargin: "0px",
+    }
+  );
+
+  const [forwardTarget, setForwardTarget] = useState<
+    RefObject<HTMLButtonElement>
+  >(allRefs?.current?.[weekDayStepValue] ?? null);
+  const [backwardTarget, setBackwardTarget] = useState<
+    RefObject<HTMLButtonElement>
+  >(allRefs?.current?.[0] ?? null);
+
+  // triggered on schedule days scroll, sets prev arrow value to {currentIntersectingDay - 14}
+  // or the first day and next arrow value to {currentIntersectingDay + 7} or the last day
+  const updateIntersected = useCallback(
+    (lastIntersected: RefObject<HTMLButtonElement>) => {
+      const newIndex = allRefs.current.findIndex(
+        (el) => el.current?.innerText === lastIntersected.current?.innerText
+      );
+      const newTarget =
+        allRefs.current[newIndex - 1] ??
+        allRefs.current[allRefs.current.length - 1];
+      const newBackwardTarget =
+        allRefs.current[newIndex - weekDayStepSkipValue] ?? allRefs.current[0];
+      setForwardTarget(newTarget);
+      setBackwardTarget(newBackwardTarget);
+    },
+    []
+  );
+
+  const weekdays = useMemo(
+    () =>
+      generatedWeekDays({
+        isScheduleTimeshifted,
+        dayDifference,
+        liveAndFutureEvents,
+        firstScheduleDate,
+        selectedDayIndex,
+        setSelectedDayIndex,
+        allRefs,
+        updateIntersected,
+      }),
+    [
+      isScheduleTimeshifted,
+      dayDifference,
+      liveAndFutureEvents,
+      firstScheduleDate,
+      selectedDayIndex,
+      setSelectedDayIndex,
+      allRefs,
+      updateIntersected,
+    ]
+  );
+
+  const handleWeekScrollForward = () => {
+    const scrollRef =
+      forwardTarget?.current ?? allRefs?.current?.[weekDayStepValue].current;
+    scrollRef?.scrollIntoView({
+      behavior: "smooth",
+      block: "nearest",
+      inline: "start",
+    });
+    recalculateSkipPrevValues(scrollRef);
+  };
+
+  // triggered on arrow click, sets prev arrow value to {currentIntersectingDay - 14} or the first day
+  // and next arrow value to {currentIntersectingDay + 7} or the last day
```

Comment:
Comments are good, but this one like the previous and next, doesn't need to have hard-coded numbers, there are already constants for it. Best to describe the why, not the what.

Reply:
[Q] Since this comment is repeated 3 times, isn't the logic inside also repeated? Can it be easily extracted into a helper function or would it take too much time thus making it OOS?

Reply:
replaced comments + added a `TODO` [ede5a78](https://github.com/sparkletown/sparkle/pull/2921/commits/ede5a78fee222b03794b50d81d1b5bb452d05814)